sycl/doc/extensions/IntelGPU/IntelGPUDeviceInfo.md

```diff
@@ -0,0 +1,239 @@
+
+    if (dev.has(aspect::ext_intel_pci_address)) {
+      auto BDF = dev.get_info<info::device::ext_intel_pci_address>();
+    }
+
+
+
+# Intel GPU Execution Unit SIMD Width #
+
+A new device descriptor will be added which will provide the physical SIMD width of an execution unit on an Intel GPU.  This data will be used to calculate the computational capabilities of the device.
+
+This support will only be provided when using the Level Zero backend \(BE\).  This is the default for Intel GPUs.  The OpenCL BE does not provide the physical SIMD width at this time.
+
+## Device Information Descriptors ##
+
+| Device Descriptors | Return Type | Description |
+| ------------------ | ----------- | ----------- |
+| info\:\:device\:\:ext\_intel\_gpu\_eu\_simd\_width | uint32\_t| Returns the physical SIMD width of the  execution unit (EU).|
+
+
+## Aspects ##
+
+A new aspect, ext\_intel\_gpu\_eu\_simd\_width, will be added.
+
+
+## Error Condition ##
+
+An invalid object runtime error will be thrown if the device does not support aspect\:\:ext\_intel\_gpu\_eu\_simd\_width.
+
+## Example Usage ##
+
+The physical EU SIMD width can be obtained using the standard get\_info() interface.
+
+    if (dev.has(aspect::ext_intel_gpu_eu_simd_width)) {
+        auto euSimdWidth = dev.get_info<info::device::ext_intel_gpu_eu_simd_width>();
+    }
+
+
+
+# Intel GPU Execution Unit Count #
+
+A new device descriptor will be added which will provide the number of execution units on an Intel GPU.  If the device is a subdevice, then the number of EUs in the subdevice is returned.
+
+This new device descriptor will provide the same information as "max\_compute\_units" does today.  We would like to have an API which is specific for Intel GPUs.
+
+This support will only be provided when using the Level Zero backend \(BE\).  This is the default for Intel GPUs.
+
+## Device Information Descriptors ##
+
+| Device Descriptors | Return Type | Description |
+| ------------------ | ----------- | ----------- |
+| info\:\:device\:\:ext\_intel\_gpu\__eu\_count | uint32\_t| Returns the number of execution units (EUs) associated with the Intel GPU.|
+
+
+## Aspects ##
+
+A new aspect, ext\_intel\_gpu\_eu\_count, will be added.
+
+
+## Error Condition ##
+
+An invalid object runtime error will be thrown if the device does not support aspect\:\:ext\_intel\_gpu\_eu\_count.
+
+## Example Usage ##
+
+Then the number of EUs can be obtained using the standard get\_info() interface.
+
+    if (dev.has(aspect::ext_intel_gpu_eu_count)) {
+      auto euCount = dev.get_info<info::device::ext_intel_gpu_eu_count>();
+    }
+
+
+
+# Intel GPU Number of Slices #
+
+A new device descriptor will be added which will provide the number of slices on an Intel GPU.  If the device is a subdevice, then the number of slices in the subdevice is returned.
+
+This support will only be provided when using the Level Zero backend \(BE\).  This is the default for Intel GPUs.
```

Comment:
These two sentences appear for all the queries.  I was looking at this with fresh eyes, and I think it's not strictly correct.  This would probably be a better way to phrase it:

> This new device descriptor is only available for devices in the Level Zero platform, and the matching aspect (described below) is only true for those devices.

If you think it's important to say that the sentence about the default:

> This new device descriptor is only available for devices in the Level Zero platform, and the matching aspect (described below) is only true for those devices.  The DPC++ default behavior is to expose GPU devices through the Level Zero platform.

This wording seems more precise because an application can "use" more than one backend at the same time.  Therefore, not all devices in the application will support these queries even if the application uses the Level Zero backend.

Reply:
nice!